src/Commands/make/CmdCommand.php

```diff
@@ -0,0 +1,125 @@
+    public function run()
+    {
+        // Validate argument was passed
+        if (empty($this->getArguments())) {
+            $this->console->output->alert('make:cmd requires an argument.');
+            return;
+        }
+
+        $this->createCommand();
+    }
+
+    /**
+     * Create a new command.
+     *
+     * Gets the provided arguments and creates subdirectories if they do not already exist
+     * for the provided command. The last element in the chain of subdirectories is treated
+     * as the name of the callabale command, in which we create a template command file for.
+     *
+     * For example:
+     *
+     *      php blacksmith make:cmd this:is:my:command
+     *
+     * results in the following path being created:
+     *
+     *      my-project/commands/this/is/my/Command.php
+     */
+    protected function createCommand()
+    {
+        $arg = $this->getArguments()[0];
+
+        $sub_dir_names = explode(DIRECTORY_SEPARATOR, $arg);
+        $command_name  = $sub_dir_names[count($sub_dir_names) - 1];
+
+        // Remove the command name from the list of sub dirs
+        unset($sub_dir_names[count($sub_dir_names) - 1]);
+
+        $this->createSubDirectories($sub_dir_names);
+        $this->createCommandFile($command_name);
+    }
+
+    /**
+     * Create the subdirectories for the new command.
+     *
+     * @param array $sub_dirs
+     *      The array of subdirectory names to create for the new command.
+     */
+    protected function createSubDirectories(array $sub_dirs)
+    {
+        // Get the path to the /commands folder
+        $blacksmith_cmd_dir = BLACKSMITH_ROOT . BLACKSMITH_COMMANDS_DIR;
+
+        // Form the full path with sub dirs for the new command
+        // e.g. "my-project/commands/my/new/Cmd.php"
+        $this->command_dir_path = $blacksmith_cmd_dir . DIRECTORY_SEPARATOR . implode(DIRECTORY_SEPARATOR, $sub_dirs);
+
+        if (!file_exists($this->command_dir_path)) {
+            mkdir($this->command_dir_path, 0755, true);
+        }
+    }
+
+    /**
+     * Create the .php file for the new command.
+     *
+     * @param string $command_name
+     *      The name of the new command to create the command file for.
+     */
+    protected function createCommandFile($command_name)
+    {
+        // Check if the command already has the .php extension
+        $file_extension = strtolower(pathinfo($command_name, PATHINFO_EXTENSION));
+
+        if ($file_extension !== 'php') {
+            $command_name .= '.php';
+        }
+
+        $command_file_name      = ucfirst($command_name);
+        $command_file_full_path = $this->command_dir_path . DIRECTORY_SEPARATOR . $command_file_name;
+
+        // Check if the command already exists
+        if (file_exists($command_file_full_path)) {
+            $this->console->output->alert('Command already exists.');
+            return;
+        }
+
+        // Create the file for the new command
+        fopen($command_file_full_path, 'w');
```

Comment:
So I looked into `fopen` a little bit more and learned that `fopen` only generates a warning if it fails to write and that an opened file will get closed at script termination. With that in mind, it still is best practice to use `fclose` to close it manually, so I'll be adding that fix.

I also wrapped that in a try/catch and will be throwing an error if the command being created already exists, which will be caught so an alert message can be printed to the console.